### PR TITLE
Functional tests -  Fix wait for navigation and disable category tests

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Fix wait for navigation and disable category tests
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | no tests needed

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19749)

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]
Looks fine to me

File: tests/UI/pages/BO/catalog/categories/index.js [hunk 1]
```diff
@@ -332,7 +332,7 @@ module.exports = class Categories extends BOBasePage {
     let i = 0;
     while (await this.elementNotVisible(sortColumnDiv, 500) && i < 2) {
       await this.page.hover(this.sortColumnDiv(sortBy));
-      await this.clickAndWaitForNavigation(sortColumnSpanButton);
+      await this.clickAndWaitForNavigation(sortColumnSpanButton, 'newtorkidle');
```
Comment (SimonGrn) [inline tests/UI/pages/BO/catalog/categories/index.js]:
```suggestion
      await this.clickAndWaitForNavigation(sortColumnSpanButton, 'networkidle');
```

File: tests/UI/pages/commonPage.js [hunk 2]
```diff
@@ -243,11 +243,12 @@ module.exports = class CommonPage {
   /**
    * Go to Page and wait for navigation
    * @param selector
+   * @param waitUntil
    * @return {Promise<void>}
    */
-  async clickAndWaitForNavigation(selector) {
+  async clickAndWaitForNavigation(selector, waitUntil = 'load') {
```
Comment (SimonGrn) [inline tests/UI/pages/commonPage.js:249]: It's the default setting.

File: tests/UI/pages/commonPage.js [hunk 3]
```diff
@@ -243,11 +243,12 @@ module.exports = class CommonPage {
...
     await Promise.all([
-      this.page.waitForNavigation(),
+      this.page.waitForNavigation(waitUntil),
```
Comment (SimonGrn) [inline tests/UI/pages/commonPage.js]: What is `waitUntil` here ? I don't see it in the method signature

Comment (boubkerbribri) [reply tests/UI/pages/commonPage.js]: It's the navigation event to wait, I will add it in docs.

Comment (SimonGrn) [reply tests/UI/pages/commonPage.js]: Oh yes I needed to open the full file to see it, It wan't visible on the review page :/

**Review by PierreRambaud — DISMISSED** [review]

File: tests/UI/pages/commonPage.js (hunk 2 above)
Comment (PierreRambaud) [reply tests/UI/pages/commonPage.js:249]: Yes, but it aims to be optional in the method content

Comment (boubkerbribri) [reply tests/UI/pages/commonPage.js:249]: yes, it's default setting for [waitUntil](https://playwright.dev/#version=v1.1.1&path=docs%2Fapi.md&q=pagewaitfornavigationoptions) , and optional here

**Review by NeOMakinG — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Pull request merged**